syntax = "proto3";

package cart;


service Cart {
  rpc GetCart (GetCartRequest) returns (GetCartResponse);
  rpc AddToCart (AddToCartRequest) returns (AddToCartResponse);
  rpc DeleteFromCart (DeleteFromCartRequest) returns (DeleteFromCartResponse);
  rpc ClearCart (ClearCartRequest) returns (ClearCartResponse);
  rpc UpdateCart (UpdateCartRequest) returns (UpdateCartResponse);
}


message GetCartRequest {
}

message ProductCartEntry {
  int32 quantity = 1;
  string item_id = 2;
}

message GetCartResponse {
  repeated ProductCartEntry cart = 1;
}

message AddToCartRequest {
  string product_id = 2;
  int32 quantity = 3;
}

message AddToCartResponse {
}

message DeleteFromCartRequest {
  string product_id = 2;
}

message DeleteFromCartResponse {
}

message ClearCartRequest {
}

message ClearCartResponse {
}

message UpdateCartRequest {
  string product_id = 2;
  int32 quantity = 3;
}

message UpdateCartResponse {
}
